### PR TITLE
Add Stack Destructor

### PR DESCRIPTION
**Pull request by cube2222**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (non-breaking change that adds documentation)

## Related issues

> Fix [#1]()

### Development

- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review

- [x] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [x] Pull Request is no longer marked as "draft"
- [x] Reviewers have been assigned
- [ ] Changes have been reviewed by at least one other engineer

### CODE REVIEW

**Review by michieldewilde — COMMENTED** [review]
lgtm

**Review by b4k3r — APPROVED** [review]
And readme? Will you add later?

File: spacelift/resource_stack_destructor.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+}
+
+func resourceStackDestructorDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	if deactivated := d.Get("deactivated"); deactivated != nil && deactivated.(bool) {
+		d.SetId("")
+		return nil
+	}
+
+	var mutation struct {
+		DeleteStack *structs.Stack `graphql:"stackDelete(id: $id, destroy: $destroy)"`
```
Comment (marcinwyszynski) [inline spacelift/resource_stack_destructor.go:83]: I don't think you need to pass "destroy" as a variable here?

Comment (marcinwyszynski) [reply spacelift/resource_stack_destructor.go:83]: Also, should it not be called `destroyResources`?

Comment (cube2222) [reply spacelift/resource_stack_destructor.go:83]: I don't think I understand what you mean, and it's called destroy.

Comment (cube2222) [reply spacelift/resource_stack_destructor.go:83]: Ah, I can pass the const directly, right

Comment (cube2222) [reply spacelift/resource_stack_destructor.go]: fixed

**Comment by cube2222** [comment]
> And readme? Will you add later?

Yes, when I'm sure it works.

**Review by marcinwyszynski — APPROVED** [review]
🤞🏻

**Pull request merged**